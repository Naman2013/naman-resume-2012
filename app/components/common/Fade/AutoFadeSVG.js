import React from 'react';
import PropTypes from 'prop-types';

const AutoFadeSVG = ({ children, duration }) => (
  <g
    className="auto-fade-svg"
    style={{
      animation: `FADE-IN ${duration}s infinite`,
    }}
  >
    {children}
    <style jsx>{`
      @keyframes FADE-IN {
        0%   { opacity: 0; }
        100% { opacity: 1; }
      }
    `}
    </style>
  </g>
);

AutoFadeSVG.propTypes = {
  children: PropTypes.node.isRequired,
  duration: PropTypes.number,
};

AutoFadeSVG.defaultProps = {
  duration: 5,
};

export default AutoFadeSVG;
