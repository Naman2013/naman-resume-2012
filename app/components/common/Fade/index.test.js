import React from 'react';
import { shallow } from 'enzyme';
import AutoFadeSVG from './AutoFadeSVG';

describe('AutoFadeSVG', () => {
  const autoFadeSVG = shallow(<AutoFadeSVG><g className="test-container" /></AutoFadeSVG>);

  it('should render correctly', () => {
    expect(autoFadeSVG).toMatchSnapshot();
  });

  it('should render children', () => {
    expect(autoFadeSVG.find('.test-container').length).toEqual(1);
  });
});
