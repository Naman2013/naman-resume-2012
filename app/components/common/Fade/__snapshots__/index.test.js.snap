// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AutoFadeSVG should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AutoFadeSVG
    duration={5}
  >
    <g
      className="test-container"
    />
  </AutoFadeSVG>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <g
          className="test-container"
        />,
        <JSXStyle
          css={
            Array [
              ".auto-fade-svg.__jsx-style-dynamic-selector{-webkit-animation:FADE-IN-__jsx-style-dynamic-selector 5s;animation:FADE-IN-__jsx-style-dynamic-selector 5s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-out;animation-timing-function:ease-out;}",
              "@-webkit-keyframes FADE-IN-__jsx-style-dynamic-selector{0%{opacity:0;}100%{opacity:1;}}",
              "@keyframes FADE-IN-__jsx-style-dynamic-selector{0%{opacity:0;}100%{opacity:1;}}",
            ]
          }
          dynamic={
            Array [
              5,
            ]
          }
          styleId="3211801733"
        />,
      ],
      "className": "jsx-559850450 auto-fade-svg",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "test-container",
        },
        "ref": null,
        "rendered": null,
        "type": "g",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": Array [
            ".auto-fade-svg.__jsx-style-dynamic-selector{-webkit-animation:FADE-IN-__jsx-style-dynamic-selector 5s;animation:FADE-IN-__jsx-style-dynamic-selector 5s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-out;animation-timing-function:ease-out;}",
            "@-webkit-keyframes FADE-IN-__jsx-style-dynamic-selector{0%{opacity:0;}100%{opacity:1;}}",
            "@keyframes FADE-IN-__jsx-style-dynamic-selector{0%{opacity:0;}100%{opacity:1;}}",
          ],
          "dynamic": Array [
            5,
          ],
          "styleId": "3211801733",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "g",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <g
            className="test-container"
          />,
          <JSXStyle
            css={
              Array [
                ".auto-fade-svg.__jsx-style-dynamic-selector{-webkit-animation:FADE-IN-__jsx-style-dynamic-selector 5s;animation:FADE-IN-__jsx-style-dynamic-selector 5s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-out;animation-timing-function:ease-out;}",
                "@-webkit-keyframes FADE-IN-__jsx-style-dynamic-selector{0%{opacity:0;}100%{opacity:1;}}",
                "@keyframes FADE-IN-__jsx-style-dynamic-selector{0%{opacity:0;}100%{opacity:1;}}",
              ]
            }
            dynamic={
              Array [
                5,
              ]
            }
            styleId="3211801733"
          />,
        ],
        "className": "jsx-559850450 auto-fade-svg",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "test-container",
          },
          "ref": null,
          "rendered": null,
          "type": "g",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": Array [
              ".auto-fade-svg.__jsx-style-dynamic-selector{-webkit-animation:FADE-IN-__jsx-style-dynamic-selector 5s;animation:FADE-IN-__jsx-style-dynamic-selector 5s;-webkit-animation-fill-mode:forwards;animation-fill-mode:forwards;-webkit-animation-timing-function:ease-out;animation-timing-function:ease-out;}",
              "@-webkit-keyframes FADE-IN-__jsx-style-dynamic-selector{0%{opacity:0;}100%{opacity:1;}}",
              "@keyframes FADE-IN-__jsx-style-dynamic-selector{0%{opacity:0;}100%{opacity:1;}}",
            ],
            "dynamic": Array [
              5,
            ],
            "styleId": "3211801733",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "g",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
